op-node/rollup/derive/batches_test.go

```diff
@@ -539,6 +558,42 @@ func TestValidBatch(t *testing.T) {
 			},
 			Expected: BatchDrop, // dropped because it could have advanced the epoch to B
 		},
+		{
+			Name:       "transactions in Isthmus upgrade block",
+			L1Blocks:   []eth.L1BlockRef{l1A, l1B, l1C},
+			L2SafeHead: l2A0,
+			Batch: BatchWithL1InclusionBlock{
+				L1InclusionBlock: l1B,
+				Batch: &SingularBatch{
+					ParentHash:   l2A1.ParentHash,
+					EpochNum:     rollup.Epoch(l2A1.L1Origin.Number),
+					EpochHash:    l2A1.L1Origin.Hash,
+					Timestamp:    l2A1.Time,
+					Transactions: []hexutil.Bytes{[]byte("invalid tx in isthmus upgrade block")},
+				},
+			},
+			Expected:    BatchDrop,
+			ExpectedLog: "dropping batch with user transactions in fork activation block",
+			ConfigMod:   isthmusAt(&l2A1.Time),
+		},
+		{
+			Name:       "transactions in Interop upgrade block",
+			L1Blocks:   []eth.L1BlockRef{l1A, l1B, l1C},
```

Comment:
I wonder if we can create these automatically so we pick up new forks.  We have the list of hard forks already and we could iterate through them, then for each check that they forbid user transactions unless they're in a list that explicitly allows it.  Then if we add a new fork, as long as we add it to the list of forks this test will remind us to update the if statement in `batches.go` and prevent user transactions (or add it to the allow list if it doesn't have upgrade tx).  Otherwise I can see us forgetting that.

Reply:
Good idea, I was thinking of actually running the whole upgrade block action tests for all forks from Isthmus. But then, as you can see with the commented out code, it just failed for Interop, so I didn't follow up with this idea. Will do this in the batches tests and track it for Interop.